src/internal/FluencePeer.ts

```diff
@@ -0,0 +1,350 @@
+            await this._connection.disconnect();
+        }
+        this._clearWathcDog();
+        this._requests.forEach((r) => {
+            r.cancel();
+        });
+    }
+
+    private get _selfPeerId(): PeerIdB58 {
+        return this._selfPeerIdFull.toB58String();
+    }
+
+    private get _relayPeerId(): PeerIdB58 | undefined {
+        return this._connection?.nodePeerId.toB58String();
+    }
+
+    private async _executeIncomingParticle(particle: Particle) {
+        logParticle(log.debug, 'external particle received', particle);
+
+        let request = this._requests.get(particle.id);
+        if (request) {
+            await request.receiveUpdate(particle);
+        } else {
+            request = RequestFlow.createExternal(particle);
+            request.handler.combineWith(this._callServiceHandler);
+        }
+        this._requests.set(request.id, request);
+
+        await this._processRequest(request);
+    }
+
+    private _processRequest(request: RequestFlow) {
+        try {
+            this._currentRequestId = request.id;
+            request.execute(this._interpreter, this._connection, this._relayPeerId);
+        } catch (err) {
+            log.error('particle processing failed: ' + err);
+        } finally {
+            this._currentRequestId = null;
+        }
+    }
+
+    private _interpreterCallback: ParticleHandler = (
+        serviceId: string,
+        fnName: string,
+        args: any[],
+        tetraplets: SecurityTetraplet[][],
+    ): CallServiceResult => {
+        if (this._currentRequestId === null) {
+            throw Error('current request can`t be null here');
+        }
+
+        const request = this._requests.get(this._currentRequestId);
+        const particle = request.getParticle();
+        if (particle === null) {
+            throw new Error("particle can't be null here");
+        }
+        const res = request.handler.execute({
+            serviceId,
+            fnName,
+            args,
+            tetraplets,
+            particleContext: {
+                particleId: request.id,
+                initPeerId: particle.init_peer_id,
+                timeStamp: particle.timestamp,
+                ttl: particle.ttl,
+                signature: particle.signature,
+            },
+        });
+
+        if (res.result === undefined) {
+            log.error(
+                `Call to serviceId=${serviceId} fnName=${fnName} unexpectedly returned undefined result, falling back to null`,
+            );
+            res.result = null;
+        }
+
+        return {
+            ret_code: res.retCode,
+            result: JSON.stringify(res.result),
+        };
+    };
+
+    private _initWatchDog() {
+        this._watchDog = setInterval(() => {
+            for (let key in this._requests.keys) {
+                if (this._requests.get(key).hasExpired()) {
+                    this._requests.delete(key);
+                }
+            }
+        }, 5000);
```

Comment:
Let's add todo to make `5000` configurable

Reply:
ok